src/app/backing-up-from-results-not-allowed-guard.ts

```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, RouterStateSnapshot } from '@angular/router';
+@Injectable()
+export class BackingUpFromResultsNotAllowedGuard implements CanActivate {
+
+  previousPageResultsOrUndefined = true;
+
+  constructor() { }
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean {
+    const access = !this.previousPageResultsOrUndefined || route.url.toString().indexOf('kysymykset') === -1;
```

Comment:
Can you not handle this with _only_ the provided router snapshots? That way canActivate is a pure function.